src/Domain/BlockNeighbor.cpp

```diff
@@ -0,0 +1,70 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/BlockNeighbor.hpp"
+
+#include <algorithm>
+#include <ostream>
+
+template <size_t VolumeDim>
+BlockNeighbor<VolumeDim>::BlockNeighbor(
+    const size_t id, const Orientation<VolumeDim>& orientation)
+    : id_(id), orientation_ptr_{orientation.get_clone()} {}
+
+template <size_t VolumeDim>
+BlockNeighbor<VolumeDim>::BlockNeighbor(
+    const BlockNeighbor<VolumeDim>& neighbor)
+    : id_(neighbor.id_),
+      orientation_ptr_{neighbor.orientation_ptr_->get_clone()} {}
+
+template <size_t VolumeDim>
+BlockNeighbor<VolumeDim>& BlockNeighbor<VolumeDim>::operator=(
+    const BlockNeighbor<VolumeDim>& rhs) {
+  if (&rhs == this) {
+    return *this;
+  }
+  this->id_ = rhs.id_;
+  this->orientation_ptr_ = std::move(rhs.orientation_ptr_->get_clone());
+  return *this;
+}
+
+template <size_t N>
+bool operator==(const BlockNeighbor<N>& lhs, const BlockNeighbor<N>& rhs) {
+  return lhs.id_ == rhs.id_ and lhs.orientation_ptr_ == rhs.orientation_ptr_;
```

Comment:
This should presumably compare the Orientation objects, rather than the addresses.